sample_structure.py

```diff
@@ -79,10 +77,17 @@ def load_and_write_surfaces(self):
         for bulk_ind, bulk in enumerate(self.all_bulks):
             possible_surfaces = bulk.get_possible_surfaces()
             if self.args.enumerate_all_structures:
-                self.logger.info(f'Enumerating all {len(possible_surfaces)} surfaces for bulk {self.bulk_indices_list[bulk_ind]}')
-                for surface_ind, surface_info in enumerate(possible_surfaces):
-                    surface = Surface(bulk, surface_info, surface_ind, len(possible_surfaces))
-                    self.combine_and_write(surface, self.bulk_indices_list[bulk_ind], surface_ind)
+                if self.args.surface_index is not None:
```

Comment:
Just to clarify the API, this means that enumerate-all-structures CLA has to be provided even when we are passing a single surface index?

Reply:
Right now that flag is needed so that the script knows to enumerate all adslab configs. But that's irrelevant until `combine_and_write()`, so you have a good point - I don't need that nested if statement for creating the Structure object.

Reply:
Maybe it makes sense to refactor a bit then? 
I think adding the different use-cases that are supported from this script would be useful (as comments for now in the bottom of the script.)

Reply:
Actually, I think what I have right now is needed because having --enumerate_all_structures changes some things down the line (e.g. directory naming `[adsorbate_ind]_[bulk_ind]_[surface_ind]_adslab[config_ind]` vs `random[seed]_adslab`), so I will leave this as is.

There could be cleaner ways of organizing this if we want to add the functionality to specify any of {adsorbate, bulk, surface} indices and randomly sample the rest without enumerating all structures. But I think this refactoring can be done later / separately.